src/Model/HasAttributesTrait.php

```diff
@@ -171,6 +190,12 @@ public function setDocumentAttribute(string $key, $value)
             $value = $this->{$this->buildMutatorMethod($key, 'set')}($value);
         }
 
+        if ($this->shouldCastUTCDateTime($key, $value)) {
+            $this->attributes[$key] = Caster::castToMongoUTCDateTime($value);
+
+            return $this->attributes[$key];
```

Comment:
Dessa forma, ele mantem o comportamento de limpar o campo quando for nulo.

```suggestion
            $value = Caster::castToMongoUTCDateTime($value);
```